src/util/googleMaps.js

```diff
@@ -55,6 +92,33 @@ export const getPlaceDetails = (placeId, sessionToken) =>
     });
   });
 
+/**
+ * Fetches detailed information about a specific place using the new Google Maps Places API.
+ *
+ * @param {string} placeId - ID for a place received from the
+ * autocomplete service
+ * @returns {Promise<Object|undefined>} A promise that resolves to an object containing:
+ *   - `adress` (string): The formatted address of the place.
+ *   - `origin` (object): The geographic origin of the place (calculated using `placeOriginNew`).
+ *   - `bounds` (object): The viewport bounds of the place (calculated using `placeBoundsNew`).
+ */
+export const getPlaceDetailsNew = async placeId => {
+  try {
+    const place = await new window.google.maps.places.Place({ id: placeId });
+    const fields = ['addressComponents', 'formattedAddress', 'viewport', 'id', 'location'];
+
+    await place.fetchFields({ fields: fields });
+
+    return {
+      adress: place.formattedAddress,
+      origin: placeOriginNew(place),
+      bounds: placeBoundsNew(place),
+    };
+  } catch (error) {
+    console.error(`Could not get details for place id "${placeId}": `, error);
```

Comment:
I don't think we need to fill our logs with these error messages.

What you could do is to add this to the beginning of the file:
`const isDev = process.env.NODE_ENV === 'development';`

and conditionally print the error:

```diff
if (isDev) {
  console.error(`Could not get details for place id "${placeId}": `, error);
}
```

Reply:
Actually, since this is marked to be an async function. You should return a rejected promise:

```js
return Promise.reject(error);
```